Tutorials/CNTK_302B_Image_Super-resolution_Using_CNNs_and_GANs.ipynb

```diff
@@ -0,0 +1,1928 @@
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Select the right target device when this notebook is being tested:\n",
+    "if 'TEST_DEVICE' in os.environ:\n",
+    "    if os.environ['TEST_DEVICE'] == 'cpu':\n",
+    "        C.device.try_set_default_device(C.device.cpu())\n",
+    "    else:\n",
+    "        C.device.try_set_default_device(C.device.gpu(0))\n",
+    "C.device.try_set_default_device(C.device.gpu(0))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "collapsed": true,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [],
+   "source": [
+    "isFast = True"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "<p>There are two run modes:\n",
+    "<ul>\n",
+    "<li><i>Fast mode:</i> <code>isFast</code> is set to True. This is the default mode for the notebooks, which means we train for fewer iterations or train/test on limited data. This ensures functional correctness of the notebook though the models produced are far from what a completed training would produce.</li>\n",
+    "<li><i>Slow mode</i>: We recommend the user to set this flag to False once the user has gained familiarity with the notebook content and wants to gain insight from running the notebooks for a longer period with different parameters for training.</li>\n",
+    "</ul></p>\n",
+    "\n",
+    "<p><b>Note</b> If the <code>isFlag</code> is set to False the notebook will take several days on a GPU enabled machine. You can try fewer iterations by setting the <code>NUM_MINIBATCHES</code> to a smaller number which comes at the expense of quality of the generated images. You can also try reducing the <code>MINIBATCH_SIZE</code>.</p>\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "## Training configuration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "deletable": true,
+    "editable": true
+   },
+   "source": [
+    "<p>We will use minibatches of size 64. Lower the minibatch size if you are getting <i>CUDA out of memory</i> error. Model is trained on 64 x 64 input patches prepared from <b>Berkeley Segementation Dataset (BSDS300)</b> (folders <b>train64_LR</b> and <b>train64_HR</b>). Learning rate will be decreasing during training and that is visible in the computation graph section.</p>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {
+    "collapsed": false,
+    "deletable": true,
+    "editable": true
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Data directory is Data/BerkeleySegmentationDataset\n"
+     ]
+    }
+   ],
+   "source": [
+    "#training configuration\n",
+    "MINIBATCH_SIZE = 64\n",
+    "NUM_MINIBATCHES = 1000 if isFast else 200000\n",
```

Comment:
change the fast mode to 200 iterations. Also, set NUM_MINIBATCHES for the entire notebook here. There are other instances in the notebook that needs to be removed for the isFast flag to be true.